tests/models/test_param.py

```diff
@@ -272,7 +273,24 @@ def return_num(num):
         ti = dr.get_task_instances()[0]
         assert ti.xcom_pull() == 'test'
 
-    def test_param_non_json_serializable(self):
+    @pytest.mark.parametrize(
+        'default, should_warn',
+        [
+            pytest.param(
+                {0, 1, 2},
+                True,
+                id='default-non-JSON-serializable',
+            ),
+            pytest.param(None, False, id='default-None'),  # Param init should not warn
+            pytest.param({"b": 1}, False, id='default-JSON-serializable'),  # Param init should not warn
+        ],
+    )
+    def test_param_json_warning(self, default, should_warn):
+        warning_msg = 'The use of non-json-serializable params is deprecated'
```

Comment:
Can you also use this in the second context manager (currently line 295)?

```python
with pytest.warns(DeprecationWarning, match=warning_msg):
```

Reply:
yup meant to do that thanks